plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal.hs

```diff
@@ -291,23 +301,67 @@ fieldSuggestCodeAction recorder ide _ (CodeActionParams _ _ (TextDocumentIdentif
 -- TODO: Support more definitions than sections.
 gotoDefinition :: PluginMethodHandler IdeState LSP.Method_TextDocumentDefinition
 gotoDefinition ideState _ msgParam = do
-    nfp <- getNormalizedFilePathE uri
-    cabalFields <- runActionE "cabal-plugin.commonSections" ideState $ useE ParseCabalFields nfp
-    case CabalFields.findTextWord cursor cabalFields of
-      Nothing ->
-        pure $ InR $ InR Null
-      Just cursorText -> do
-        commonSections <- runActionE "cabal-plugin.commonSections" ideState $ useE ParseCabalCommonSections nfp
-        case find (isSectionArgName cursorText) commonSections of
-          Nothing ->
-            pure $ InR $ InR Null
-          Just commonSection -> do
-            pure $ InL $ Definition $ InL $ Location uri $ CabalFields.getFieldLSPRange commonSection
-    where
-      cursor = Types.lspPositionToCabalPosition (msgParam ^. JL.position)
-      uri = msgParam ^. JL.textDocument . JL.uri
-      isSectionArgName name (Syntax.Section _ sectionArgName _) = name == CabalFields.onelineSectionArgs sectionArgName
-      isSectionArgName _ _ = False
+  nfp <- getNormalizedFilePathE uri
+  cabalFields <- runActionE "cabal-plugin.commonSections" ideState $ useE ParseCabalFields nfp
+  case CabalFields.findTextWord cursor cabalFields of
+    Nothing ->
+      pure $ InR $ InR Null
+    Just cursorText -> do
+      commonSections <- runActionE "cabal-plugin.commonSections" ideState $ useE ParseCabalCommonSections nfp
+      case find (isSectionArgName cursorText) commonSections of
+        Nothing ->
+          pure $ InR $ InR Null
+        Just commonSection -> do
+          pure $ InL $ Definition $ InL $ Location uri $ CabalFields.getFieldLSPRange commonSection
+  where
+    cursor = Types.lspPositionToCabalPosition (msgParam ^. JL.position)
+    uri = msgParam ^. JL.textDocument . JL.uri
+    isSectionArgName name (Syntax.Section _ sectionArgName _) = name == CabalFields.onelineSectionArgs sectionArgName
+    isSectionArgName _ _ = False
+
+-- | Handler for hover messages.
+--
+-- Provides a Handler for displaying message on hover.
+-- If found that the filtered hover message is a dependency,
+-- adds a Documentation link.
+hover :: PluginMethodHandler IdeState LSP.Method_TextDocumentHover
+hover ide _ msgParam = do
+  nfp <- getNormalizedFilePathE uri
+  (cabalFields, _) <- runActionE "cabal.cabal-hover" ide $ useWithStaleE ParseCabalFields nfp
```

Comment:
I think `useWithStaleE` doesn't quite work with `cabal` files, and we had some bad UX with it in the past.
One reason it doesn't work that well is that we do not do any Position Mapping iirc, so outdated cabal files will have wrong positions... However, hover is usually used when the language server has settled, so it might not be noticeable at all.

Reply:
I think it's a good idea to change it to the `useWithStale`, so it will be used more often in the future, but I thought I used it excessively in my other solutions.

Should I change it up there as well? Or it was a good choice earlier?

Reply:
Rewrote to the `useWithStale` for hover.

Reply:
`useWithStale` is the same thing but without explicit error handling. I think `useE` is the best function.

Reply:
Ah, I didn't understand that the issue was with `withStale` part and not with error handling. I will change it back then, to `useE`.